css/ast/src/selector.rs

```diff
@@ -12,7 +12,16 @@ pub struct SelectorList {
 #[ast_node("ComplexSelector")]
 pub struct ComplexSelector {
     pub span: Span,
-    pub selectors: Vec<CompoundSelector>,
+    pub children: Vec<ComplexSelectorChildren>,
+}
+
+#[ast_node]
+#[derive(Is)]
+pub enum ComplexSelectorChildren {
```

Comment:
What's the purpose of this enum?

Reply:
I think this change may make implementing transforms harder.

Reply:
Because combinator is not a part of compound selector, it is part of complex selector, compound selector can be written with/without combinator, in theory in future it will be possible to write:
```
.class {
  > .foo {
      color: red;
  }
}

.class > {
  .foo {
      color: red;
  }
}
```

note sass and less already support it, because they parse it as part of `complex` selector.

Look at grammar https://www.w3.org/TR/selectors-4/#grammar:
```
<complex-selector> = <compound-selector> [ <combinator>? <compound-selector> ]*
<compound-selector> = [ <type-selector>? <subclass-selector>*
                        [ <pseudo-element-selector> <pseudo-class-selector>* ]* ]!
```

as you can see complex selector has children:
- `compound-selector`
- `combinator` (optional)

Reply:
Then this is fine, thanks!